src/target/riscv/program.c

```diff
@@ -48,7 +48,7 @@ int riscv_program_exec(struct riscv_program *p, struct target *t)
 	if (riscv_program_ebreak(p) != ERROR_OK) {
 		LOG_TARGET_ERROR(t, "Unable to insert ebreak into program buffer");
 		for (size_t i = 0; i < riscv_progbuf_size(p->target); ++i)
-			LOG_TARGET_ERROR(t, "ram[%02x]: DASM(0x%08" PRIx32 ") [0x%08" PRIx32 "]",
+			LOG_TARGET_ERROR(t, "progbuf[%02x]: DASM(0x%08" PRIx32 ") [0x%08" PRIx32 "]",
 					(int)i, p->progbuf[i], p->progbuf[i]);
```

Comment:
```suggestion
			LOG_TARGET_ERROR(t, "progbuf[%02zx]: DASM(0x%08" PRIx32 ") [0x%08" PRIx32 "]",
					i, p->progbuf[i], p->progbuf[i]);
```
Though IMHO `riscv_progbuf_size()` should return `unsigned int`, not `size_t`
1. There is a hard cap of 16 (for some weird reason `RISCV_MAX_PROGBUF_SIZE` is 32, I've thought this is from 0.11 spec but `riscv_program_exec` is not implemented by 0.11 targets).
2. `size_t` is form host context, `riscv_progbuf_size()` is from target context. To be completely correct, when referring to a size of a memory area something like `target_addr_t` should be used instead (IMHO `target_size_t` would be even better but there is none currently).

Reply:
Thanks for the feedback.

> en-sc: for some weird reason RISCV_MAX_PROGBUF_SIZE is 32 [...]

Addressed:

- Constant renamed to `RISCV013_MAX_PROGBUF_SIZE` and value changed to `16`.
- Constants `RISCV_REGISTER_COUNT` and `RISCV_DSCRATCH_COUNT` were removed - they are not used anywhere.

> en-sc: riscv_progbuf_size() should return unsigned int, not size_t

Addressed:

- Return type of riscv_progbuf_size() changed `size_t` -> `unsigned int`
- The type of `instruction_count` in `struct riscv_program` changed the same way
- Few statements in program.c updated accordingly